src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -1071,29 +1038,48 @@ namespace
 
         try
         {
-            if (extension == ".exr")
-            {
-                write_exr_image(
-                    bf_file_path,
-                    image,
-                    image_attributes);
+            bool high_dynamic_range_format = false;
+            if (extension == ".exr" || extension == ".tiff" || extension == ".tif" || extension == ".hdr")
+                high_dynamic_range_format = true;            
+
+            create_parent_directories(bf_file_path);
+
+            const std::string filename = bf_file_path.string();
+            GenericImageFileWriter writer(filename.c_str());
+
+            // Add image to writer
+            Image *transformed_image;
```

Comment:
You have a memory leak since you're never deleting the transformed image. Please use `std::unique_ptr<>` to avoid this problem.

Reply:
Oh, I misunderstand the origin code. I will fix this serious problem.